library/ssl_srv.c

```diff
@@ -3165,18 +3164,11 @@ static int ssl_write_server_key_exchange( mbedtls_ssl_context *ssl )
             return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
         }
 
-        MBEDTLS_SSL_DEBUG_BUF( 3, "parameters hash", hash, hashlen != 0 ? hashlen :
-            (unsigned int) ( mbedtls_md_get_size( mbedtls_md_info_from_type( md_alg ) ) ) );
+        MBEDTLS_SSL_DEBUG_BUF( 3, "parameters hash", hash, hashlen );
 
         /*
          * 3.3: Compute and add the signature
          */
-        if( mbedtls_ssl_own_key( ssl ) == NULL )
-        {
-            MBEDTLS_SSL_DEBUG_MSG( 1, ( "got no private key" ) );
-            return( MBEDTLS_ERR_SSL_PRIVATE_KEY_REQUIRED );
-        }
-
```

Comment:
What change are you refering to? Note that the test has not been removed, it's just moved a bit further down. Do you see a problem in the non-async case with the change order?

If the async callback gives up or is not present, then this test does run, just before calling `mbedtls_pk_sign`. This was the reason to move it further down.

Reply:
The change I was refering to was the removal of this block in this commit. Now that you mention it, I can see that it indeed present later, but that was done in a previous commit, which is why I didn't make the connection. So now that I see the complete picture, I agree that the logic is fine.

However, I think it would have made review easier if that block was moved in a single commit, rather than duplicated in one commit and then the old copy removed in a later commit. Anyway, thanks for clarifying.

Reply:
My bad, I must have made a mistake when making partial commits or rebasing. Do you want to me to rewrite the history? I'd rather not, because it would disrupt the reviews.

Reply:
We're on the same line here, I also think (quite strongly) that rewriting history in the middle of the review process would be too disruptive.

On the other hand, I have no strong opinion on whether cleaning up history _after_ the PR was fully reviewed, in a way that keeps the total diff unchanged, is worth the effort or not. If we think it is, then this would indeed be a good candidate for history cleanup.